docs/run-platform-server/ingestion.mdx

```diff
@@ -0,0 +1,92 @@
+- if your application can work on a [filtered network dataset](./ingestion-filtering.mdx) based on specific accounts and assets, then we recommend applying ingestion filter rules. When using filter rules, it provides benefit of choice in longer historical retention timeframe since the filtering is reducing the overall database size to such a degree, historical retention(`HISTORY_RETENTION_COUNT`) can be set in terms of years rather than months or even disabled(`HISTORY_RETENTION_COUNT=0`)
+- if you cannot limit your history retention window to 30 days and cannot use filter rules, we recommend considering [Stellar Hubble Data Warehouse](https://developers.stellar.org/docs/accessing-data/overview) for any historical data
+
+### Ingesting Live Data
+
+This option is enabled by default and is the recommended mode of ingestion to run. It is controlled with environment configuration flag `INGEST`. Refer to [Configuration](./configuring.mdx) for how an instance of Horizon performs the ingestion role.
+
+For a high availability requirements, **we recommend deploying more than one live ingesting instance**, as this makes it easier to avoid downtime during upgrades and adds resilience, ensuring you always have the latest network data, refer to [Ingestion Role Instance](./configuring.mdx#multiple-instance-deployment)
+
+### Ingesting Historical Data
+
+Import network data from a past date range in to the database:
+
+<CodeExample>
+
+```
+stellar-horizon db reingest range <start> <end>
+```
+
+</CodeExample>
+
+Running any historical range of ingestion requires coordination with the data retention configuration chosen. When setting a temporal limit on history with `HISTORY_RETENTION_COUNT=<LEDGER_COUNT>`, the temporal limit takes precedence, and any data ingested beyond that limit will be automatically purged.
+
+Typically the only time you need to run historical ingestion is once when boot-strapping a system after first deployment, from that point forward **live** ingestion will keep the database populated with the expected sliding window of trailing historical data. Maybe one exception is if you think you have a gap in the database caused by the **live** ingestion being down, in which case you can run historical ingestion range to essentially gap fill.
+
+You can run historical ingestion in parallel in background while your main Horizon server separately performs **live** ingestion. If the range specified overlaps with data already in the database, it is ok and will simply be overwritten, effectively idempotent.
+
+#### Parallel ingestion workers
+
+You can break up historical date range into slices and run each in parallel as a separate process:
+
+<CodeExample>
+
+```
+horizon1> stellar-horizon db reingest range 1 10000
```

Comment:
ok, we could weave the usage in like so:


> You can parallelize the ingestion of one target historical date range by dividing it into sequential slices of smaller ranges and run the `db reingest range` command for each sub range in parallel as a separate process on the same or a different machine. The shorthand rule for best performance is to identify the number of CPU cores available per target machine, if multi-core, then add `--parallel-workers <num of cores>` to the command, this will enable the command to further parallelize internally down to smaller ranges. 

<CodeExample>

```
# target range 1 30000, on single machine with 1 CPU core
horizon1> stellar-horizon db reingest range 1 30000

# target range 1 30000, on single machine with 4 CPU cores
horizon1> stellar-horizon db reingest range 1 30000 --parallel-workers --parallel-workers 4

# target range 1 30000, on single machine with 4 CPU cores
horizon1> stellar-horizon db reingest range 1 15000 --parallel-workers 2
horizon1> stellar-horizon db reingest range 15001 30000 --parallel-workers 2

# target range 1 30000, on two machines, each has 4 CPU cores
horizon1> stellar-horizon db reingest range 1 15000 --parallel-workers 4
horizon2> stellar-horizon db reingest range 15001 30000 --parallel-workers 4
```

</CodeExample>




`

Reply:
ok, cleaned up the example per suggestion, thanks!